src/oauthcli/flow.py

```diff
@@ -302,9 +282,8 @@ def auth_server(
         local_server.timeout = timeout_seconds
         local_server.handle_request()
 
-        # Note: using https here because oauthlib is very picky that
-        # OAuth 2.0 should only occur over https.
-        authorization_response = wsgi_app.last_request_uri.replace("http", "https")
+        # OAuth 2.0 should only occur over https, disable oauthlib security
+        os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = "true"
```

Comment:
Why? Seems suspicious :)

Reply:
It seem to the only official way to disable the oauthlib lib security: https://oauthlib.readthedocs.io/en/latest/oauth2/security.html. In fact, your previous solution does not works during my tests, so I switch to this official solution.

Reply:
During my tests to add Forgejo auth provider I finally catched and fixed what you seen there :)